ion-hash/src/representation.rs

```diff
@@ -57,13 +67,20 @@ impl<'a, D> Update for EscapingDigest<'a, D>
 where
     D: Update,
 {
+    /// Escapes various markers as per the spec. Allocates a temporary array to
+    /// do so.
     fn update(&mut self, data: impl AsRef<[u8]>) {
+        let mut escaped = vec![];
+
         for byte in data.as_ref() {
             if let Markers::B | Markers::E | Markers::ESC = *byte {
-                self.0.update([0x0C]);
+                escaped.extend(&[Markers::ESC, *byte]);
+            } else {
+                escaped.extend(&[*byte]);
             }
-            self.0.update([*byte]);
         }
+
+        self.0.update(escaped);
```

Comment:
I'm not very happy with this. It's not obvious, but the tests made me do this (so I can see the full update sequence and skip trailing zeros).